scripts/db/src/get-user.ts

```diff
@@ -50,48 +64,100 @@ async function getByEmail(email: string, callback: DbScriptCallback) {
       POSTGRES_HOST,
       POSTGRES_DATABASE,
       POSTGRES_PORT,
+      MONGO_URI,
+      MONGO_DB_NAME,
     } = (configuration as unknown) as DbConfiguration
 
-    /** Declare connection info */
-    const connectionInfo: ConnectionConfig = {
-      user: POSTGRES_USERNAME,
-      password: POSTGRES_PASSWORD,
-      host: POSTGRES_HOST,
-      database: POSTGRES_DATABASE,
-      port: POSTGRES_PORT ? parseInt(POSTGRES_PORT) : 5432,
-      ssl: TEMPLATE_DATA.pgShouldSsl,
-    }
+    /**
+     * Logic in this function tries to match:
+     * EAForum/packages/lesswrong/server/vulcan-lib/apollo-server/authentication.tsx
+     */
+    async function getForumUser(): Promise<CallbackUser | null> {
+      // Connect to the Forum DB
+      const mongoClient: MongoClient = new MongoClient(MONGO_URI)
+      await mongoClient.connect()
+
+      // Query the users collection for someone with our email
+      const forumUser = await mongoClient
+        .db(MONGO_DB_NAME)
+        .collection<ForumUser>('users')
+        .findOne({ 'emails.address': email })
+
+      await mongoClient.close()
+
+      if (!forumUser) {
+        return null
+      }
+
+      // Which email did we find?
+      const emailInfo = forumUser.emails.find((e) => e.address === email)
+      if (!emailInfo) {
+        // This should never happen, as they were returned by mongo because that
```

Comment:
If this is basically logically constrained from happening, is it worth having the error condition? Alternatively, are you so sure that this could never happen (e.g. if the Mongo search is case-insensitive, the `.find()` call could still fail to get a match.

Reply:
Find is case-sensitive. I do think generally I'd rather make the type system happy by handling errors I'm sure can never happen, rather than by type assertions, because of base rates about being wrong about "this can never happen" comments.